templates/user/user-about.html

```diff
@@ -67,10 +130,122 @@ <h4>Rating History</h4>
+                $div.find('#year').attr('data-year', year);
+                $div.find('#prev-year-action').css('display', year > metadata.min_year ? '' : 'none');
+                $div.find('#next-year-action').css('display', year < current_year ? '' : 'none');
+                var current_weekday = 0;
+                var start_day = new Date(year, 0, 1)
+                var end_day = new Date(year + 1, 0, 0);
+                if (year == current_year) {
+                    end_day = new Date();
+                    start_day = new Date(end_day.getFullYear() - 1, end_day.getMonth(), end_day.getDate() + 1);
+                    $div.find('#year').text("{{ _('past year') }}");
+                } else {
+                    $div.find('#year').text(year);
+                }
+                var days = [];
+                for (var day = start_day, day_num = 1; day <= end_day; day.setDate(day.getDate() + 1), day_num++) {
+                    var isodate = day.toISOString().split('T')[0];
+                    days.push({
+                        date: new Date(day),
+                        weekday: day.getDay(),
+                        day_num: day_num,
+                        activity: submission_activity[isodate] || 0,
+                    });
+                }
+
+                var sum_activity = days.map(obj => obj.activity).reduce((a, b) => a + b, 0);
+                // There doesn't seem to be a way to internationalize strings in JavaScript.
+                $div.find('#submission-total-count').text(
+                    "{{ _('%%(cnt)d total submissions') }}".replace('%(cnt)d', sum_activity)
+                )
+
+                var max_activity = Math.max.apply(null, days.map(obj => obj.activity));
+                var diff = max_activity / (activity_levels - 1);
+                var activity_breakdown = [...Array(activity_levels).keys()].map(x => diff * x);
+
+                for (; current_weekday < days[0].weekday; current_weekday++) {
+                    $div.find('#submission-' + current_weekday)
+                        .append($('<td></td>').addClass('activity-blank').append('<div></div>'));
+                }
+
+                days.forEach(obj => {
+                    var level = activity_breakdown.findIndex(x => x >= obj.activity);
+                    // There doesn't seem to be a way to internationalize strings in JavaScript.
+                    var text = "{{ _('%%(cnt)d submissions on %%b %%d, %%Y') }}"
+                        .replace('%(cnt)d', obj.activity)
+                        .replace('%b', metadata.month_names[obj.date.getMonth()])
+                        .replace('%d', obj.date.getDate())
+                        .replace('%Y', obj.date.getFullYear());
+                    $div.find('#submission-' + obj.weekday)
+                        .append(
+                            $('<td></td>').addClass(['activity-label', 'activity-' + level])
```

Comment:
Pretty sure this works (here and elsewhere)

```suggestion
                            $('td').addClass(['activity-label', 'activity-' + level])
```

Reply:
It seems this selects the `td` objects already on the page and adds the class, while we want to create a new `td` object.

Reply:
The correct code is `$('<td>')`.

Reply:
Done